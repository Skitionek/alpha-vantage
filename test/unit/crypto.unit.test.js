
import AlphaVantageMock from "../mocks/alphaVantageMock";
import { matchesSnapshot } from "../jest.extensions";

jest.setTimeout(30000);

let alpha;
beforeAll(() => {
	alpha = new AlphaVantageMock();
});

it(`daily data works`, () => {
	return alpha.crypto.daily('btc', 'usd')
		.then(matchesSnapshot);
});

it(`exchangeRates data works`, () => {
	return alpha.crypto.exchangeRates('btc', 'usd')
		.then(matchesSnapshot);
});

it(`exchangeTimeSeries data works`, () => {
	return alpha.crypto.exchangeTimeSeries({
		from_symbol: 'btc',
		to_symbol: 'usd',
		interval: 'daily'
	})
		.then(matchesSnapshot);
});

it(`weekly data works`, () => {
	return alpha.crypto.weekly('btc', 'usd')
		.then(matchesSnapshot);
});

it(`monthly data works`, () => {
	return alpha.crypto.monthly('btc', 'usd')
		.then(matchesSnapshot);
});
