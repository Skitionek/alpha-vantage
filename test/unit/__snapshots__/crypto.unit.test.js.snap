// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`daily data works 1`] = `
Object {
  "data": Array [
    Object {
      "date": "Timestamp/Date",
      "from_close": "Float",
      "from_high": "Float",
      "from_low": "Float",
      "from_open": "Float",
      "market_cap": "Float",
      "to_close": "Float",
      "to_high": "Float",
      "to_low": "Float",
      "to_open": "Float",
      "volume": "Float",
    },
  ],
  "digital_currency_code": "String",
  "digital_currency_name": "String",
  "information": "String",
  "last_refreshed": "Timestamp/Date",
  "market_code": "String",
  "market_name": "String",
  "time_zone": "String",
}
`;

exports[`exchangeRates data works 1`] = `
Object {
  "ask_price": "Float",
  "bid_price": "Float",
  "exchange_rate": "Float",
  "from_currency_code": "String",
  "from_currency_name": "String",
  "last_refreshed": "Timestamp/Date",
  "time_zone": "String",
  "to_currency_code": "String",
  "to_currency_name": "String",
}
`;

exports[`exchangeTimeSeries data works 1`] = `
Object {
  "data": Array [
    Object {
      "date": "Timestamp/Date",
      "from_close": "Float",
      "from_high": "Float",
      "from_low": "Float",
      "from_open": "Float",
      "market_cap": "Float",
      "to_close": "Float",
      "to_high": "Float",
      "to_low": "Float",
      "to_open": "Float",
      "volume": "Float",
    },
  ],
  "digital_currency_code": "String",
  "digital_currency_name": "String",
  "information": "String",
  "last_refreshed": "Timestamp/Date",
  "market_code": "String",
  "market_name": "String",
  "time_zone": "String",
}
`;

exports[`monthly data works 1`] = `
Object {
  "data": Array [
    Object {
      "date": "Timestamp/Date",
      "from_close": "Float",
      "from_high": "Float",
      "from_low": "Float",
      "from_open": "Float",
      "market_cap": "Float",
      "to_close": "Float",
      "to_high": "Float",
      "to_low": "Float",
      "to_open": "Float",
      "volume": "Float",
    },
  ],
  "digital_currency_code": "String",
  "digital_currency_name": "String",
  "information": "String",
  "last_refreshed": "Timestamp/Date",
  "market_code": "String",
  "market_name": "String",
  "time_zone": "String",
}
`;

exports[`weekly data works 1`] = `
Object {
  "data": Array [
    Object {
      "date": "Timestamp/Date",
      "from_close": "Float",
      "from_high": "Float",
      "from_low": "Float",
      "from_open": "Float",
      "market_cap": "Float",
      "to_close": "Float",
      "to_high": "Float",
      "to_low": "Float",
      "to_open": "Float",
      "volume": "Float",
    },
  ],
  "digital_currency_code": "String",
  "digital_currency_name": "String",
  "information": "String",
  "last_refreshed": "Timestamp/Date",
  "market_code": "String",
  "market_name": "String",
  "time_zone": "String",
}
`;
